src/omp_compat.h

```diff
@@ -25,11 +25,30 @@
 // so this is what LAMMPS primarily uses.  For those compilers
 // that strictly implement OpenMP 4.0 (such as GCC 9.0 and later
 // or Clang 10.0 and later), we give up default(none).
+
+// autodetect OpenMP compatibility if not explicitly set
+
+#ifndef LAMMPS_OMP_COMPAT
+#  if defined(__INTEL_COMPILER)
+#    if __INTEL_COMPILER > 18
+#      define LAMMPS_OMP_COMPAT 4
+#    endif
+#  elif defined(__clang__)
+#    if __clang_major__ >= 10
+#      define LAMMPS_OMP_COMPAT 4
+#    endif
+#  elif defined(__GNUC__)
+#    if __GNUC__ >= 0
```

Comment:
Wasn't there this whole problem with older gcc version? (https://github.com/lammps/lammps/pull/1330#issuecomment-463373306)

Reply:
Maybe I was wrong and that was just for OpenMP 5 and gcc-9.

Reply:
No this is a typo and it should be 9 instead of 0. It just didn't show in the integration testing, since all tests with newer compilers seem to be using CMake and there the define is already set by CMake. The purpose of this is primarily to have less errors that require explaining when people use the legacy build system, now that Linux distributions with newer compilers are becoming more common.